pkg/util/kube/kube.go

```diff
@@ -426,8 +486,8 @@ func deleteJob(ctx context.Context, client kubernetes.Interface, job *batchv1.Jo
 }
 
 func getPodByJob(ctx context.Context, client kubernetes.Interface, job *batchv1.Job) (*v1.Pod, error) {
-	pods, err := client.CoreV1().Pods(store.Get().DefaultNamespace).List(ctx, metav1.ListOptions{
-		LabelSelector: "controller-uid=" + job.GetLabels()["controller-uid"],
+	pods, err := client.CoreV1().Pods(job.GetNamespace()).List(ctx, metav1.ListOptions{
```

Comment:
Does this returns the latest pod created by this job?

Reply:
yeah, i think every time the job creates a new pod, it uses a different controller-uid (and updates the job with it)
i'll make sure